src/job.rs

```diff
@@ -1266,20 +1260,9 @@ pub(crate) async fn load_next(
     context: &Context,
     thread: Thread,
     info: &InterruptInfo,
-) -> Option<Job> {
+) -> Result<Option<Job>> {
     info!(context, "loading job for {}-thread", thread);
 
-    while !context.sql.is_open().await {
-        // The db is closed, which means that this thread should not be running.
-        // Wait until the db is re-opened (if we returned None, this thread might do further damage)
-        warn!(
-            context,
-            "{}: load_next() was called but the db was not opened, THIS SHOULD NOT HAPPEN. Waiting...",
-            thread
-        );
-        sleep(Duration::from_millis(500)).await;
-    }
-
```

Comment:
i just had a closed look: this db-really-opened-precaution was added at https://github.com/deltachat/deltachat-core-rust/pull/2255 because of an possible database-corruption-issue.  

and that can be removed now as all uis were changed to really call stopIo() before imex(), right?

Reply:
It was not needed even after #2255 merge. The real fix in #2255 is the resultification of `houskeeping()`, so it quits if it can't enumerate used blobs instead of removing all of them as if none are used. As long as all the functions that query the database don't ignore the errors, there is no need to check if the database is open explicitly.

Reply:
cc @Hocuri 
But I don't see why it is needed when everything is resultified.

Reply:
thanks for taking care and for clarification!